abook.py

```diff
@@ -396,4 +411,27 @@ def vcf2abook() -> None:
     )
     args = parser.parse_args()
 
-    Abook.abook_file(args.infile, args.outfile)
+    
+    if os.path.isfile(args.outfile):
+        abook = Abook(args.outfile)
+        abook_uids = abook.get_uids()
+        full_names = []
+        # get full names of incoming vcf contact
+        for line in args.infile:
+            if re.search(re.compile("^FN:.?"), line):
+                full_name = line.strip("FN:").strip()
+                full_names.append(full_name)
+
+        # check full name against all abook contacts (by uid)
+        for uid in abook_uids:
+            name = abook.get_fn_by_uid(uid)
+        
+            if name in full_names:
+                print(name, "is already in addressbook. Skipping")
+            else:
+                abook.append_vobject(args.infile)
```

Comment:
this looks wrong, the first argument of `append_vobject` wants type `Component` whereas `args.infile` is a file object.